### PR TITLE
feat: use security-group module instead of resource

### PR DESCRIPTION
**Pull request by SweetOps**
## what
* use security-group module instead of resource

## why
* more flexible  than current implementation 
* bring configuration of security group/rules to one standard
* bump minimal required terraform version to 0.14

## references
* CPCO-409

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -296,27 +296,16 @@ resource "aws_iam_instance_profile" "ec2" {
...
-  }
-
-  tags = module.this.tags
+module "default_sg" {
+  source          = "cloudposse/security-group/aws"
+  version         = "0.1.2"
+  use_name_prefix = var.security_group_use_name_prefix
+  rules           = var.security_group_rules
+  vpc_id          = var.vpc_id
+  description     = "EFS Security Group"
```
Comment (Gowiem) [inline main.tf:305]: Looks like an incorrect description -- let's update.

**Review by nitrocode — APPROVED** [review]

**Pull request merged**